src/main/java/hudson/plugins/plot/pipeline/PlotBuilder.java

```diff
@@ -0,0 +1,203 @@
+
+    public String getYaxisMinimum() { return yaxisMinimum; }
+
+    public String getYaxisMaximum() { return yaxisMaximum; }
+
+    public List<Series> getSeries() { return series; }
+
+    public List<CSVSeries> getCsvSeries() { return csvSeries; }
+
+    public List<PropertiesSeries> getPropertiesSeries() { return propertiesSeries; }
+
+    public List<XMLSeries> getXmlSeries() { return xmlSeries; }
+
+    @Override
+    public void perform(Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener) {
+        plots = new ArrayList<>();
+        Plot plot = new Plot(title, yaxis, group, numBuilds, csvFileName, style, false, false, false, false, yaxisMinimum, yaxisMaximum);
+        plot.series = series;
+        plot.addBuild(build, listener.getLogger(), workspace);
+        plots.add(plot);
+        PlotBuildAction buildAction = build.getAction(PlotBuildAction.class);
+        if (buildAction == null) {
+            build.addAction(new PlotBuildAction(build, plots));
+        } else {
+            buildAction.addPlots(plots);
+        }
+    }
+
+    // Overridden for better type safety.
+    // If your plugin doesn't really define any property on Descriptor,
+    // you don't have to do this.
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl)super.getDescriptor();
+    }
+
+    /**
+     * Descriptor for {@link PlotBuilder}. Used as a singleton.
+     * The class is marked as public so that it can be accessed from views.
+     */
+    @Extension // This indicates to Jenkins that this is an implementation of an extension point.
+    @Symbol("plot")
+    public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
+        /*
+         * To persist global configuration information,
+         * simply store it in a field and call save().
+         *
+         * <p>
+         * If you don't want fields to be persisted, use <tt>transient</tt>.
+         */
+
+        /**
+         * In order to load the persisted global configuration, you have to
+         * call load() in the constructor.
+         */
+        public DescriptorImpl() {
+            load();
+        }
+
+        public String getCsvFileName() {
+            return "plot-" + String.valueOf( (int)Math.round( Math.random() * 100000000 ) ) + ".csv";
+        }
+
+        public FormValidation doCheckName(@QueryParameter String value)
+                throws IOException, ServletException {
+            if (value.length() == 0)
+                return FormValidation.error("Please set a group");
+            if (value.length() < 4)
+                return FormValidation.warning("Isn't the group too short?");
+            return FormValidation.ok();
+        }
+
+        public boolean isApplicable(Class<? extends AbstractProject> aClass) {
+            // Indicates that this builder can be used with all kinds of project types 
+            return false;
```

Comment:
🐛 I think you mean to return false, if you want it to work with all AbstractProject types.